app/components/viewer/Viewer3D.tsx

```diff
@@ -1,20 +1,39 @@
-import React, { useContext } from 'react';
+import React, { Suspense, useContext, useRef } from 'react';
 import * as THREE from 'three';
-import { Canvas } from '@react-three/fiber';
+import { Canvas, useLoader, useThree } from '@react-three/fiber';
 import { OrbitControls, Sky } from '@react-three/drei';
 import { ReplayData } from '../../lib/api/apiRequests';
 import { ReplayLines } from './ReplayLines';
+import { TimeLine, TimeLineInfos } from './TimeLine';
 import { Grid, DEFAULT_GRID_POS } from './Grid';
 import { SettingsContext } from '../../lib/contexts/SettingsContext';
+import FrameRate from './FrameRate';
+import ReplayCars from './ReplayCars';
 
 const BACKGROUND_COLOR = new THREE.Color(0.05, 0.05, 0.05);
 
 interface Props {
     replaysData: ReplayData[];
 }
 
+const timeLineGlobal = new TimeLineInfos();
+
 const Viewer3D = ({ replaysData }: Props): JSX.Element => {
-    const { lineType, showGearChanges } = useContext(SettingsContext);
+    const {
+        lineType,
+        showGearChanges,
+        showFPS,
+        showInputOverlay,
+        replayLineOpacity,
+        replayCarOpacity,
+        cameraMode,
+    } = useContext(SettingsContext);
+    const orbitControlsRef = useRef<any>();
+
+    let orbitDefaultTarget = DEFAULT_GRID_POS;
+    if (timeLineGlobal.currentRaceTime === 0 && replaysData.length > 0) {
+        orbitDefaultTarget = replaysData[0].samples[0].position;
```

Comment:
`samples` is pretty much always non-empty, but might be safe to have a check here?